src/query/boolean_query/boolean_query.rs

```diff
@@ -19,6 +20,65 @@ use std::collections::BTreeSet;
 /// `MustNot` occurence.
 /// * match at least one of the subqueries that is not
 /// a `MustNot` occurence.
+///
+/// ```rust
+///#[macro_use]
+///extern crate tantivy;
+///use tantivy::collector::Count;
+///use tantivy::query::QueryParser;
+///use tantivy::schema::{Schema, TEXT};
+///use tantivy::{Index, Result};
+///
+///fn main() -> Result<()> {
+///    let mut schema_builder = Schema::builder();
+///    let title = schema_builder.add_text_field("title", TEXT);
+///    let schema = schema_builder.build();
+///    let index = Index::create_in_ram(schema);
+///    {
+///        let mut index_writer = index.writer(3_000_000)?;
+///        index_writer.add_document(doc!(
+///            title => "The Name of the Wind",
+///        ));
+///        index_writer.add_document(doc!(
+///            title => "The Diary of Muadib",
+///        ));
+///        index_writer.add_document(doc!(
+///            title => "A Dairy Cow",
+///        ));
+///        index_writer.add_document(doc!(
+///            title => "The Diary of a Young Girl",
+///        ));
+///        index_writer.commit().unwrap();
+///    }
+///
+///    let reader = index.reader()?;
+///    let searcher = reader.searcher();
+///
+///    let query_parser = QueryParser::for_index(&index, vec![title]);
```

Comment:
This looks like a doctest for query parser than for `BooleanQuery`.

In my experience, users have a hard time understanding how to compose subqueries into another one using  `BooleanQuery`.

For instance, having an app with a search field and a facet system that makes it possible to drill down.

That requires composign the output of the query parser with the filtering facet `TermQuery`.